### PR TITLE
77 profile redesign

### LINKED ISSUE #77: Profile (match history) - Mobile Responsiveness
CSS needs to be updated to break to mobile layout sooner when shrinking the page so that player cards don't start overflowing.

### PR DESCRIPTION
**Pull request by skspade**
Hoping to have an a simple and easy win here. 

- Increases the break point into mobile layout for xl screens instead of md screens
- also wrapped fetch functions into useCallbacks and added them to the deps array to calm the "missing deps" warning

closes #77

### CODE REVIEW

**Review by erin-fitzpatric — CHANGES_REQUESTED** [review]
I'm nervous this might be breaking to "mobile" view too easily. It might be that the layout just needs to be redesigned. This is at half size on my desktop, and probably similar to many laptops:
image

File: src/middleware.ts
Comment (erin-fitzpatric) [inline src/middleware.ts:1]: Does this middleware get applied to all routes? I want the profile stats to update in "like real time" aka the 3 minutes that we check the leaderboard API. Players check these after their games so they need to update ASAP.

**Comment by erin-fitzpatric** [comment]
image

**Comment by skspade** [comment]
localhost_3000_profile_1073800087 Overview (2024-09-28 06 54 18)
localhost_3000_profile_1073800087 Overview (2024-09-28 06 54 33)

**Comment by erin-fitzpatric** [comment]
LOVE IT

**Review by erin-fitzpatric — APPROVED** [review]

**Pull request merged**